src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {
+  FC,
+  memo,
+  useCallback,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoTitleField } from '../TodoTitleField/TodoTitleFields';
+
+  type Props = {
+    todo: Todo,
+    isDeleting: boolean,
+    onDeleteTodo: (todoId: number) => void,
+    updateTodo: (
+      todoId: number,
+      fieldsToUpdate: Partial<Pick<Todo, 'title' | 'completed'>>
+    ) => Promise<void>,
+    isUpdating: boolean,
+  };
+
+export const TodoItem: FC<Props> = memo(({
+  todo,
+  isDeleting,
+  onDeleteTodo,
+  updateTodo,
+  isUpdating,
+}) => {
+  const isLoading = todo.id === 0 || isDeleting || isUpdating;
+
+  const [shouldShowInput, setShouldShowInput] = useState(false);
+
+  const cancelEditing = useCallback(() => setShouldShowInput(false), []);
+
+  const updateTitle = useCallback(async (title: string) => {
+    await updateTodo(todo.id, { title });
+  }, [todo.id]);
+
+  const deleteTodoById = useCallback(async () => {
+    await onDeleteTodo(todo.id);
+  }, [todo.id]);
+
+  return (
+    <div
+      data-cy="Todo"
+      className={cn('todo', { completed: todo.completed })}
+      key={todo.id}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          readOnly
+          onClick={() => updateTodo(todo.id, { completed: !todo.completed })}
+        />
+      </label>
+      {shouldShowInput
+        ? (
+          <TodoTitleField
+            deleteTodoById={deleteTodoById}
+            updateTitle={updateTitle}
+            cancelEditing={cancelEditing}
```

Comment:
Props and callback should be named properly
```suggestion
          <TodoTitleField
            onDeleteTodoById={handleDeleteTodoById}
            onUpdateTitle={handleUpdateTitle}
            onCancelEditing={handleCancelEditing}
```